lib/src/internal/mod.rs

```diff
@@ -1,30 +1,56 @@
 use once_cell::sync::Lazy;
 use rustc_version_runtime::version;
-use serde_json::{Value, json};
+use serde::Serialize;
 use std::io::{Error};
 use local_handler::LocalHandler;
-use voidstar_handler::{VoidstarHandler};
+use voidstar_handler::VoidstarHandler;
+use noop_handler::NoOpHandler;
 
 mod local_handler;
+mod noop_handler;
 mod voidstar_handler;
 
+#[derive(Serialize, Debug)]
+struct AntithesisLanguageInfo {
+    name: &'static str,
+    version: String,
+}
+
+#[derive(Serialize, Debug)]
+struct AntithesisVersionInfo {
+    language: AntithesisLanguageInfo,
+    sdk_version: &'static str,
+    protocol_version: &'static str,
+}
+
+#[derive(Serialize, Debug)]
+struct AntithesisSDKInfo {
+    antithesis_sdk: AntithesisVersionInfo,
+}
+
 // Hardly ever changes, refers to the underlying JSON representation
 const PROTOCOL_VERSION: &str = "1.0.0";
 
 // Tracks SDK releases
-const SDK_VERSION: &str = "0.1.1";
+const SDK_VERSION: &str = "0.1.2";
 
 pub(crate) static LIB_HANDLER: Lazy<Box<dyn LibHandler + Sync + Send>> = Lazy::new(|| {
     let handler: Box<dyn LibHandler + Sync + Send> = match VoidstarHandler::try_load() {
         Ok(handler) => Box::new(handler),
-        Err(_) => Box::new(LocalHandler::new()),
+        Err(_) => match LocalHandler::new() {
+            Some(h) => Box::new(h),
+            None => Box::new(NoOpHandler::new())
+        }
+
     };
-    let _ = handler.output(&sdk_info());
+    let s = serde_json::to_string(&sdk_info()).unwrap_or("{}".to_owned());
```

Comment:
Should we produce a better error event (instead of the empty object `{}`) here?
Or maybe add a TODO comment for now?